File: ios/Freedom/Core/Parser/TextMessageParser.swift
// ═════════════════════════════════════════════════════════════════════════════
//  TextMessageParser — Simple user text messages
// ═════════════════════════════════════════════════════════════════════════════

import Foundation

enum TextMessageParser {

    static func parse(_ raw: String, senderAddress: String? = nil) -> ParsedMessage? {
        var content = raw
        if content.hasPrefix("TEXT:") {
            content = String(content.dropFirst(5))
        }
        guard !content.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else { return nil }
        return ParsedMessage(type: .TEXT, content: content, senderAddress: senderAddress)
    }
}

File: main/src/ui/java/freedom/app/parser/TextMessageParser.kt
package freedom.app.parser

object TextMessageParser {

    private const val PREFIX = "TEXT:"

    fun matches(raw: String): Boolean = raw.startsWith(PREFIX) || raw.isNotBlank()

    fun parse(raw: String, senderAddress: String? = null): ParsedMessage {
        val content = if (raw.startsWith(PREFIX)) raw.removePrefix(PREFIX).trim() else raw.trim()
        return ParsedMessage(MessageType.TEXT, content, senderAddress)
    }
}
